extension/source/background/TransactionsController.ts

```diff
@@ -27,78 +24,76 @@ export default class TransactionsController {
-
-            if (lastWin.width !== undefined && lastWin.left !== undefined) {
-              left = lastWin.left + lastWin.width - popupWidth - 20;
+      const promptAction = new Promise<
+        TransactionStatus.APPROVED | TransactionStatus.REJECTED
+      >((resolve, _reject) => {
+        (async () => {
+          const lastWin = await windows.getLastFocused();
+
+          const popupWidth = 400;
+          let left: number | undefined;
+
+          if (lastWin.width !== undefined && lastWin.left !== undefined) {
+            left = lastWin.left + lastWin.width - popupWidth - 20;
+          }
+
+          const popup = await windows.create({
+            url: runtime.getURL(`confirm.html?id=${id}`),
+            type: 'popup',
+            width: popupWidth,
+            height: 700,
+            left,
+          });
+
+          cleanupTasks.push(() => {
+            if (popup.id !== undefined) {
+              windows.remove(popup.id);
             }
+          });
 
-            const popup = await windows.create({
-              url: runtime.getURL(`confirm.html?id=${id}`),
-              type: 'popup',
-              width: popupWidth,
-              height: 700,
-              left,
-            });
-
-            cleanupTasks.push(() => {
-              if (popup.id !== undefined) {
-                windows.remove(popup.id);
-              }
-            });
-
-            function onRemovedListener(windowId: number) {
-              if (windowId === popup.id) {
-                resolve(undefined);
-              }
+          function onRemovedListener(windowId: number) {
+            if (windowId === popup.id) {
+              resolve(TransactionStatus.REJECTED);
             }
+          }
 
-            windows.onRemoved.addListener(onRemovedListener);
-
-            cleanupTasks.push(() => {
-              windows.onRemoved.removeListener(onRemovedListener);
-            });
+          windows.onRemoved.addListener(onRemovedListener);
 
-            function messageListener(message: unknown) {
-              if (getPropOrUndefined(message, 'id') !== id) {
-                return;
-              }
+          cleanupTasks.push(() => {
+            windows.onRemoved.removeListener(onRemovedListener);
+          });
 
-              resolve(getPropOrUndefined(message, 'result') as string);
+          function messageListener(message: unknown) {
+            if (!isType(message, PromptMessage) || message.id !== id) {
+              return;
             }
 
-            runtime.onMessage.addListener(messageListener);
-          })();
-        },
-      );
+            resolve(message.result);
+          }
+
+          runtime.onMessage.addListener(messageListener);
+        })();
+      });
 
       promptAction.finally(() => cleanupTasks.run());
       return promptAction;
     },
 
     createTransaction: async ({ params }) => {
-      const id = uuidv4().toString();
+      const id = RandomId();
```

Comment:
No need for uuid since we have `RandomId`. (You could say we're duplicating uuid rather than the other way around, but uuid is unnecessarily verbose with its hex format and dashes so... good riddance imo.)